spec/services/project_life_cycle_steps/activation_service_spec.rb

```diff
@@ -0,0 +1,450 @@
+
+        let(:service) { described_class.new(user:, project:, definitions: [definitions[0]]) }
+
+        context "with date range" do
+          let!(:phase0) { create_phase(definition: definitions[0], active: false, date_range: date - 1..date - 1) }
+
+          it "reschedules following phases" do
+            service.call(active: true)
+
+            expect(phase0.reload).to have_attributes(active: true, start_date: date - 1, finish_date: date - 1)
+            expect(phase1.reload).to have_attributes(active: true, start_date: date, finish_date: date + 1)
+            expect(phase2.reload).to have_attributes(active: true, start_date: date + 2, finish_date: date + 4)
+          end
+        end
+
+        context "when already activated" do
+          let!(:phase0) { create_phase(definition: definitions[0], active: true, date_range: date - 1..date - 1) }
+
+          it "reschedules following phases" do
+            service.call(active: true)
+
+            expect(phase0.reload).to have_attributes(active: true, start_date: date - 1, finish_date: date - 1)
+            expect(phase1.reload).to have_attributes(active: true, start_date: date, finish_date: date + 1)
+            expect(phase2.reload).to have_attributes(active: true, start_date: date + 2, finish_date: date + 4)
+          end
+        end
+
+        context "without date range" do
+          let!(:phase0) { create_phase(definition: definitions[0], active: false, date_range: nil) }
+
+          it "doesn't reschedule" do
+            service.call(active: true)
+
+            expect(phase0.reload).to have_attributes(active: true, start_date: nil, finish_date: nil)
+            expect(phase1.reload).to have_attributes(active: true, start_date: date + 2, finish_date: date + 3)
+            expect(phase2.reload).to have_attributes(active: true, start_date: date + 7, finish_date: date + 9)
+          end
+        end
+      end
+    end
+
+    context "when deactivating one phase" do
+      let!(:phase0) { create_phase(definition: definitions[0], active: true, date_range: date - 1..date - 1) }
+      let!(:phase2) { create_phase(definition: definitions[2], active: true, date_range: date + 7..date + 9) }
+
+      let(:service) { described_class.new(user:, project:, definitions: [definitions[1]]) }
+
+      context "with date range" do
+        let!(:phase1) { create_phase(definition: definitions[1], active: true, date_range: date + 2..date + 3) }
+
+        it "reschedules following phases using dates of that phase" do
+          service.call(active: false)
+
+          expect(phase0.reload).to have_attributes(active: true, start_date: date - 1, finish_date: date - 1)
+          expect(phase1.reload).to have_attributes(active: false, start_date: date + 2, finish_date: date + 3)
+          expect(phase2.reload).to have_attributes(active: true, start_date: date + 2, finish_date: date + 4)
```

Comment:
I'm wondering if the expected behaviour is to reschedule the last phase to the deactivated previous phase, or rather reschedule it to the previous active phase? In this case should it be `start_date: date` instead ?

Reply:
I did it following the spec and normally phases will already be properly following each other, so there should be no difference between using deactivated phase start date or following previous active phase. If you think this is important, we can adapt logic